pallets/simple-staking/src/lib.rs

```diff
@@ -0,0 +1,162 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+/// Edit this file to define custom logic or remove it if it is not needed.
+/// Learn more about FRAME and the core library of Substrate FRAME pallets:
+/// <https://substrate.dev/docs/en/knowledgebase/runtime/frame>
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{dispatch::DispatchResultWithPostInfo, pallet_prelude::*, inherent::Vec, traits::{Currency, ReservableCurrency}};
+	use frame_system::{pallet_prelude::*, ensure_root};
+	use frame_system::Config as SystemConfig;
+	use frame_support::sp_runtime::{RuntimeDebug};
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		/// The currency mechanism.
+		type Currency: ReservableCurrency<Self::AccountId>;
+	}
+	type BalanceOf<T> = <<T as Config>::Currency as Currency<<T as SystemConfig>::AccountId>>::Balance;
+
+
+	#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug)]
+	pub struct AuthorInfo<AccountId, Balance, BlockNumber> {
+		pub who: AccountId,
+		pub deposit: Balance,
+		pub last_block: BlockNumber
+	}
```

Comment:
```suggestion
	}
	pub type AuthorInfoOf<T> = AuthorInfo<<T as Config>::AccountId, <T as Config>::Balance, <T as frame_system::Config>::BlockNumber>;
```

and it should be helpful to use!